client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -60,6 +63,13 @@ const PatternAssembler: Step = ( { navigation, flow, stepName } ) => {
 	const siteId = useSiteIdParam();
 	const siteSlugOrId = siteSlug ? siteSlug : siteId;
 	const allPatterns = useAllPatterns();
+	const sectionPatterns = useSectionPatterns();
+
+	// Fetching the categories so they are ready when ScreenCategoryList loads
```

Comment:
The logic to fetch categories and create `sectionsMapByCategory` is here to avoid doing it in `ScreenCategoryList` because that causes a noticeable delay and the screen is blank for a moment. Any better ideas?

Reply:
Yup, I know. It's fine to simply keep it here since we will need to fetch all patterns in the future. Also, we did fetching the rendered patterns at the beginning.